src/Renderer/ThreeExtended/Feature2Mesh.js

```diff
@@ -57,230 +57,261 @@ function fillColorArray(colors, offset, length, r, g, b) {
+    const altitudes = [
+        getAltitude(options, properties, vertices),
+        getExtrude(options, properties),
+    ];
+    const colors = [
+        getColor(options, properties)];
+    colors.push(colors[0].clone());
+    colors[0].multiplyScalar(155 / 255);
+
+    const geom = prepareBufferGeometry(
+        vertices,
+        altitudes,
+        colors);
+
     const indices = [];
-    const vertices = new Float32Array(2 * 3 * coordinates.coordinates.length);
-    const colors = new Uint8Array(3 * 2 * coordinates.coordinates.length);
-    const geometry = new THREE.BufferGeometry();
-    let offset = 0;
-    let offset2 = 0;
-    let nbVertices = 0;
-    let minAltitude = Infinity;
-    /* eslint-disable-next-line */
-    for (const id in coordinates.featureVertices) {
-        // extract contour coodinates and properties of one feature
-        const { contour, property } = extractFeature(coordinates, properties, id);
-        // get altitude and extrude amount from properties
-        const altitudeBottom = getAltitude(options, property, contour);
-        minAltitude = Math.min(minAltitude, altitudeBottom);
-        const extrudeAmount = getExtrude(options, property);
-        // altitudeTopFace is the altitude of the visible top face.
-        const altitudeTopFace = altitudeBottom + extrudeAmount;
-        // add vertices of the top face
-        coordinatesToExtrudedVertices(contour, altitudeTopFace, altitudeBottom, vertices, offset2);
-        // triangulate the top face
-        nbVertices = contour.length * 3;
-        const verticesTopFace = vertices.slice(offset2, offset2 + nbVertices);
-        const triangles = Earcut(verticesTopFace, null, 3);
+    // Build indices
+    if (!multiGeomAttributes) {
+        // Single polygon case
+        const isClockWise = THREE.ShapeUtils.isClockWise(
+            vertices.slice(geometry.contour.offset,
+                geometry.contour.offset +
+                geometry.contour.count).map(c => c.xyz()));
+
+        const holes = geometry.holes.map(h => h.offset);
+        const triangles = Earcut(geom.attributes.position.array.slice(
+            vertices.length * 3), holes, 3);
         for (const indice of triangles) {
-            indices.push(offset + indice);
+            indices.push(indice);
+        }
+        addExtrudedPolygonSideFaces(
+            indices,
+            vertices.length,
+            geometry.contour,
+            isClockWise);
+        if (holes.length > 0) {
+            for (let j = 0; j < geometry.holes.length; j++) {
+                addExtrudedPolygonSideFaces(
+                    indices,
+                    vertices.length,
+                    geometry.holes[j],
+                    isClockWise);
+            }
+        }
+    } else {
+        // Multi polygon case
+        const isClockWise = THREE.ShapeUtils.isClockWise(
+            vertices.slice(
+                multiGeomAttributes[0].offset + geometry[0].contour.offset,
```

Comment:
So we assume that other geometries will have the same orientation, right? Is it safe to assume according to you?

Reply:
Yeah that's the assumption and I feel it's safe (ie: I assume that a server will always return geometries with the same orientation).